### DISCUSSION TITLE
fill depends on class Model instead of interface Arrayable

### DISCUSSION THREAD

**Discussion by whizzrd** [livewire/livewire#1958] [4. General]
using \Livewire\ComponentConcerns\InteractsWithProperties::fill with a non model object implementing Arrayable (in this case a Doctrine Entity) the fill method explicitly checks that the object is instanceof Model

Suggest checking for interface \Illuminate\Contracts\Support\Arrayable which defines toArray instead of Model to allow using fill with object that do not extend Model (i.e. Laravel-Doctrine Entities)

**Comment by iksaku** [comment]
Sounds right, we should use the `Arrayable` interface whenever we're calling `fill` method, and possibly, in some other situations as well. Will open a PR for this later on.

**Reply by iksaku** [reply]
Done! See #1976 :wink:.

**Reply by whizzrd** [reply]
Great, you explained it better than I did in #1959

**Reply by iksaku** [reply]
Wow, didn’t knew you opened a PR previous to this. But well, it seems like the new PR will cover for it. 😸
